samples/reactjs-todo/client/views/login.js

```diff
@@ -30,7 +31,7 @@ export default function Login() {
         <BackHome />
         <Card>
           <div className="cstm_form-icon  align-self-center mb-3">
-            <KeyIcon size="72px" />
+            {!state.isWebAuthn ? <KeyIcon size="72px" /> : <FingerPrintIcon size="72px" />}
```

Comment:
If I'm reading the code well, I think this is the change that necessitates the need for the global state. But, I think a better solution is to move this icon to a component that has the appropriate context for displaying the right one without the need for leaking WebAuthn state into the global space.

We could easily move this icon to the `Form` component, and keep this component clean. Or, even better still, we could move the icon to the `WebAuthn` component.

Reply:
There's no reason why icons in general have to be rendered in this top level view. If that's what's causing the struggle, let's just move them where we have the context?

Why don't we just remove them entirely from this view and put them into the `Form` component. Then the `Form` component can make all the decisions as it has the context of what should and shouldn't render without need to spread journey state into the global state. Does that make sense?

Also, the `Form` component has the information of whether the user is logging in or registering because we pass that `action.type` into the `Form` component. So, it already has the needed information to make a decision on the key versus user icon as well.

Reply:
got it! i will work on this workaround, let you know any block.